src/IoC/PersistenceExtensions.cs

```diff
@@ -110,7 +110,9 @@ public static IServiceCollection AddPersistence(this IServiceCollection services
                 .AddTransient<IRepository<LedgerPeriod, int>, EntityFrameworkRepository<LedgerPeriod, int>>(r
                     => new EntityFrameworkRepository<LedgerPeriod, int>(r.GetService<ServiceDbContext>()?.LedgerPeriods))
                 .AddTransient<IRepository<CancelledOrderDetail, int>, EntityFrameworkRepository<CancelledOrderDetail, int>>(
-                    r => new EntityFrameworkRepository<CancelledOrderDetail, int>(r.GetService<ServiceDbContext>()?.CancelledPurchaseOrderDetails));
+                    r => new EntityFrameworkRepository<CancelledOrderDetail, int>(r.GetService<ServiceDbContext>()?.CancelledPurchaseOrderDetails))
+                .AddTransient<IRepository<ShortagesEntry, string>, EntityFrameworkRepository<ShortagesEntry, string>>(r
```

Comment:
as per previous comment, Should really be an IQueryRepository<ShortagesEntry>

Reply:
Noted